index.d.ts

```diff
@@ -181,6 +181,19 @@ export interface ActionConfig<TData extends object = object> {
    * @default true
    */
   abortOnFail?: boolean;
+  /**
+   * Skip an action if this function returns a string,
+   * which is the reason it should be skipped.
+   *
+   * May also return a Promise which resolves to a string.
+   *
+   * The action will continue if action.when()
+   * returns or resolves to anything other than a string,
+   * and the return value will be ignored.
+   *
+   * @default () => true
+   */
+  when?: (data: TData) => void | string | Promise<void | string>;
```

Comment:
does this need to be renamed to `skip` as well?

Reply:
Definitely. Thanks for looking!

Reply:
Haha, I forgot to update the types file! This is as good an argument for a full TS conversion as any.